src/main/java/io/odpf/firehose/sink/prometheus/builder/TimeSeriesBuilder.java

```diff
@@ -0,0 +1,97 @@
+package io.odpf.firehose.sink.prometheus.builder;
+
+import io.odpf.firehose.config.PromSinkConfig;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+
+import cortexpb.Cortex;
+
+import java.util.Properties;
+import java.util.List;
+import java.util.Map;
+import java.util.ArrayList;
+
+import static io.odpf.firehose.sink.prometheus.PromSinkConstants.*;
+
+/**
+ * Builder for Cortex TimeSeries.
+ */
+public class TimeSeriesBuilder {
+
+    private final Properties metricNameProtoIndexMapping;
+    private final Properties labelNameProtoIndexMapping;
+    private Integer timestampIndex;
+    private Boolean isEventTimestampEnabled;
+
+    /**
+     * Instantiates a new cortex time series builder.
+     *
+     * @param config the prometheus sink config
+     */
+    public TimeSeriesBuilder(PromSinkConfig config) {
+        this(config.getSinkPromMetricNameProtoIndexMapping(), config.getSinkPromLabelNameProtoIndexMapping());
+        isEventTimestampEnabled = config.isEventTimestampEnabled();
+        timestampIndex = config.getSinkPromProtoEventTimestampIndex();
+    }
+
+    private TimeSeriesBuilder(Properties metricNameProtoIndexMapping, Properties labelNameProtoIndexMapping) {
+        this.metricNameProtoIndexMapping = metricNameProtoIndexMapping == null ? new Properties() : metricNameProtoIndexMapping;
+        this.labelNameProtoIndexMapping = labelNameProtoIndexMapping == null ? new Properties() : labelNameProtoIndexMapping;
+    }
+
+    /**
+     * build list of sorted cortex time series object.
+     *
+     * @param message   the protobuf message
+     * @param partition the kafka partition where the message is consumed
+     * @return list of sorted cortex time series object
+     * @throws InvalidProtocolBufferException the exception on invalid protobuf
+     */
+    public List<Cortex.TimeSeries> buildTimeSeries(Message message, int partition) throws InvalidProtocolBufferException {
+        Cortex.TimeSeries.Builder cortexTimeSeriesBuilder = Cortex.TimeSeries.newBuilder();
+        Cortex.LabelPair.Builder cortexLabelBuilder = Cortex.LabelPair.newBuilder();
+        Cortex.Sample.Builder cortexSampleBuilder = Cortex.Sample.newBuilder();
+
+        Map<String, Object> labelPair = TimeSeriesBuilderUtils.getLabelsFromMessage(message, labelNameProtoIndexMapping, partition);
```

Comment:
labelPair name is confusing. It's not just one pair. Can we name it labelPairs or just labels?